app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,52 @@
+class RestaurantsController < ApplicationController
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+
+  # GET /restaurants
+  def index
+    @restaurants = Restaurant.all
+
+    render json: @restaurants
+  end
+
+  # GET /restaurants/1
+  def show
+    @restaurants = Restaurant.find(params[:id])
+    render json: @restaurant
+  end
+
+  # POST /restaurants
+  def create
+    @restaurant = Restaurant.new(restaurant_params)
+
+    if @restaurant.save
+      render json: @restaurant, status: :created, location: @restaurant, notice:'Restaurant was successfully created.'
+    else
+      render json: @restaurant.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /restaurants/1
+  def update
+    if @restaurant.update(restaurant_params)
```

Comment:
Same here - executing logic inside an if statement makes it harder to debug:
Imagine having to put a breakpoint: if it'll be above the `if` statement - then it's not evaluated yet.
If it's below - the condition might not have met to begin with.

consider extracting it to:

`was_update_successful = @restaurant.update(restaurant_params)`